test/parallel_api/numeric/numeric.ops/reduce_by_segment.pass.cpp

```diff
@@ -322,103 +322,69 @@ test_flag_pred()
-#endif
-    run_test<int,    BinaryPredicate<int>,    BinaryOperation<int>>();
-#if LOG_TEST_INFO
-    std::cout << "\t" << "run_test<float,  BinaryPredicate<float>,  BinaryOperation<float>>();" << std::endl;
-#endif
-    run_test<float,  BinaryPredicate<float>,  BinaryOperation<float>>();
-#if !_PSTL_ICPX_TEST_RED_BY_SEG_BROKEN_64BIT_TYPES || ONEDPL_WORKAROUND_FOR_IGPU_64BIT_REDUCTION
-#if LOG_TEST_INFO
-    std::cout << "\t" << "run_test<double, BinaryPredicate<double>, BinaryOperation<double>>();" << std::endl;
-#endif
-    run_test<double, BinaryPredicate<double>, BinaryOperation<double>>();
-#endif
+    run_test_on_host<ValueType, BinaryPredicate, BinaryOperation>();
+    run_test_on_device<ValueType, BinaryPredicate, BinaryOperation>();
 }
 
 int
 main()
 {
-#if LOG_TEST_INFO
-#if ONEDPL_WORKAROUND_FOR_IGPU_64BIT_REDUCTION
-    std::cout << "ONEDPL_WORKAROUND_FOR_IGPU_64BIT_REDUCTION is defined to " << ONEDPL_WORKAROUND_FOR_IGPU_64BIT_REDUCTION << std::endl;
-#else
-    std::cout << "ONEDPL_WORKAROUND_FOR_IGPU_64BIT_REDUCTION is not defined" << std::endl;
-#endif
-#endif
-
 #if TEST_DPCPP_BACKEND_PRESENT
     // test with flag pred
-#if LOG_TEST_INFO
-    std::cout << "test_flag_pred<sycl::usm::alloc::device, class KernelName1, std::uint64_t>();" << std::endl;
-#endif
     test_flag_pred<sycl::usm::alloc::device, class KernelName1, std::uint64_t>();
-#if LOG_TEST_INFO
-    std::cout << "test_flag_pred<sycl::usm::alloc::device, class KernelName2, dpl::complex<float>>();" << std::endl;
-#endif
     test_flag_pred<sycl::usm::alloc::device, class KernelName2, dpl::complex<float>>();
 #endif // TEST_DPCPP_BACKEND_PRESENT
 
-#if !_PSTL_ICPX_TEST_RED_BY_SEG_BROKEN_64BIT_TYPES || ONEDPL_WORKAROUND_FOR_IGPU_64BIT_REDUCTION
-#if LOG_TEST_INFO
-    std::cout << "run_test<::std::uint64_t,       UserBinaryPredicate<::std::uint64_t>,       MaxFunctor<::std::uint64_t>>();" << std::endl;
-#endif
-    run_test<::std::uint64_t,       UserBinaryPredicate<::std::uint64_t>,       MaxFunctor<::std::uint64_t>>();
-#endif
-#if LOG_TEST_INFO
-    std::cout << "run_test<::std::complex<float>, UserBinaryPredicate<::std::complex<float>>, MaxFunctor<::std::complex<float>>>();" << std::endl;
-#endif
+    run_test<::std::uint64_t, UserBinaryPredicate<::std::uint64_t>, MaxFunctor<::std::uint64_t>>();
     run_test<::std::complex<float>, UserBinaryPredicate<::std::complex<float>>, MaxFunctor<::std::complex<float>>>();
 
-#if LOG_TEST_INFO
-    std::cout << "run_test<::std::equal_to, ::std::plus>();" << std::endl;
-#endif
-    run_test<::std::equal_to, ::std::plus>();
-#if LOG_TEST_INFO
-    std::cout << "run_test<::std::equal_to, ::std::multiplies>();" << std::endl;
-#endif
+    run_test<int, ::std::equal_to<int>, ::std::plus<int>>();
+    run_test<float, ::std::equal_to<float>, ::std::plus<float>>();
+    run_test<double, ::std::equal_to<double>, ::std::plus<double>>();
+
     // TODO investigate possible overflow
-    run_test<::std::equal_to, ::std::multiplies>();
```

Comment:
It looks like we lose `int` and `float` testing for (`equal_to`, `multiplies`) for the device.  Is this intentional?

Reply:
I did it deliberately because of suspected overflows. However, it seems the overflows are possible for each data type with almost equal probability. I have returned the coverage and created #1416 with the details to look at it separately.